### PR TITLE
Migrate Ellipsis Menu component to TS

### LINKED ISSUE #36404: Migrate @woocommerce/components/ellipsis-menu to TS
### Describe the solution you'd like

Migrate the Section component to TS

### Acceptance Criteria
- [ ] All `*.js` files are converted to `*.ts` or `*.tsx`
- [ ] Run `pnpm --filter=@woocommerce/components build` without any errors
- [ ] Run `pnpm --filter=woocommerce/client/admin ts:check` without any errors

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #36404 .

Migrate Section component to TS

### How to test the changes in this Pull Request:

1. Run `pnpm --filter=@woocommerce/storybook run storybook`
2. Go to http://localhost:6007/
3. Test `EllipsisMenu` 

### Other information:

-   [X] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [X] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by samueljseay** [comment]
@moon0326 loving to see the TS migration,

I just thought I should give a hint to save some rework for y'all. In coming months we will move to React 18 and in that React.FC and VFC are no longer needed. You should move to just typing the component normally because the return type will be inferred by TS.

e.g. 

```
const Foo: React.VFC<MyProps> = (...)
```

should just be:

```
const Foo = ({...}: MyProps) => ...
```

It will save a lot of needless migration when we have to upgrade to React 18 so thought I'd give a heads up. See this article for some brief detail: https://www.mydatahack.com/moving-away-from-fc-and-vfc/

**Comment by moon0326** [comment]
> 

Thank you for the info! That's nice to know. I'll reflect the change 👍

**Comment by chihsuan** [comment]
@moon0326 Looks like 13 checks failed. Could you take a look? Thanks!

**Comment by moon0326** [comment]
> @moon0326 Looks like 13 checks failed. Could you take a look? Thanks!

Thanks! It looks like I have to go over how the component is used in other places to correct the type. I'll get back to it soon.

**Review by ROBERLANDIO — DISMISSED** [review]
https://github.com/ROBERLANDIO/-octaedro-approved-this-pull-request.-Nice-job-joshuatf-This-is-testing-well-and-the-code-

**Comment by ROBERLANDIO** [comment]
https://github.com/ROBERLANDIO/-octaedro-approved-this-pull-request.-Nice-job-joshuatf-This-is-testing-well-and-the-code-

**Review by ROBERLANDIO — COMMENTED** [review]
https://github.com/ROBERLANDIO/-octaedro-approved-this-pull-request.-Nice-job-joshuatf-This-is-testing-well-and-the-code-

**Comment by ROBERLANDIO** [comment]
https://github.com/ROBERLANDIO/-octaedro-approved-this-pull-request.-Nice-job-joshuatf-This-is-testing-well-and-the-code-

File: packages/js/components/src/ellipsis-menu/menu-title.tsx [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
- * @param {Object} props
- * @param {Node}   props.children
- * @return {Object} -
  */
-const MenuTitle = ( { children } ) => {
-	return <div className="woocommerce-ellipsis-menu__title">{ children }</div>;
-};
 
-MenuTitle.propTypes = {
+const MenuTitle: React.FC< {
```
Comment (chihsuan) [inline packages/js/components/src/ellipsis-menu/menu-title.tsx]: @moon0326 As Sam commented, we can remove `React.FC` here and use `{ children }: { children: React.ReactNode }` instead. 🙂

**Review by chihsuan — DISMISSED** [review]
Thanks for working on this! @moon0326 Tested well and LGTM.

Just a comment for React.FC. Pre-approving

File: packages/js/components/src/ellipsis-menu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		);
+
+		return (
+			<Button
+				className={ toggleClassname }
+				onClick={ ( e: MouseEvent | KeyboardEvent ) => {
+					if ( onToggle ) {
+						onToggle( e );
+					}
+					if ( _onToggle ) {
```
Comment (rjchow) [inline packages/js/components/src/ellipsis-menu/index.tsx]: Would it be clearer if we named this "toggleHandlerOverride" instead? It was a bit confusing for me to figure out what was going on here

**Review by rjchow — DISMISSED** [review]
Minor comment, lgtm otherwise!

File: packages/js/components/src/ellipsis-menu/index.tsx (hunk 2 above)
Comment (moon0326) [reply packages/js/components/src/ellipsis-menu/index.tsx]: Good suggestion 🙏 

Updated in https://github.com/woocommerce/woocommerce/pull/36405/commits/389fd602b2f59586fee778081c4159c598b563e9

**Comment by moon0326** [comment]
@rjchow @chihsuan 

Could you approve this PR one more time, please 🙏 ? My last [commit](https://github.com/woocommerce/woocommerce/pull/36405/commits/389fd602b2f59586fee778081c4159c598b563e9) dismissed your approval :(

File: packages/js/components/src/ellipsis-menu/menu-title.tsx (hunk 1 above)
Comment (chihsuan) [reply packages/js/components/src/ellipsis-menu/menu-title.tsx]: Hey @moon0326 Would you be able to change this?

**Review by chihsuan — DISMISSED** [review]
Looks good! Left only one comment on `React.FC` type.

**Review by rjchow — DISMISSED** [review]

**Comment by moon0326** [comment]
Sorry for the delay 🙏 
@rjchow @chihsuan 

Removed React.VFC in https://github.com/woocommerce/woocommerce/pull/36405/commits/659ab05e4b747bd9772430ac5564f179d23add56

**Review by rjchow — DISMISSED** [review]

**Review by rjchow — DISMISSED** [review]

**Review by chihsuan — DISMISSED** [review]

**Review by adrianduffell — APPROVED** [review]
Your latest changes LGTM 🚀

**Pull request merged**